session.go

```diff
@@ -95,39 +91,50 @@ type Session struct {
 	version wire.Version
 	path    string
 
+	localMaxRequestID  atomic.Uint64
+	remoteMaxRequestID atomic.Uint64
+
+	requestID              *sequence
```

Comment:
Since this a sequence of IDs, I think it would be better to have a plural name like requestIDs